### PR TITLE
WIP: added new function to get one_time_correlation data from two_tim…

### PR DESCRIPTION
**Pull request by sameera2004**
…e_corrrelation

### CODE REVIEW

**Comment by ericdill** [comment]
This now has merge conflicts. It also needs a test

**Comment by sameera2004** [comment]
Sure, I will fix it and add a test to that

**Comment by sameera2004** [comment]
This PR is ready for review, added tests

File: skbeam/core/correlation.py [hunk 1]
```diff
@@ -769,3 +769,29 @@ def _validate_and_transform_inputs(num_bufs, num_levels, labels):
...
+def one_time_from_two_time(two_time_corr):
+    """
+    This will provide the one-time correlation data from two-time correlation
+    data. 
+
+    Parameters
+    ----------
+    two_time_corr : array
+        matrix of two time correlation
+        shape (number of images, number of images, number of labels(ROI))
```
Comment (ericdill) [inline skbeam/core/correlation.py]: I think your shape is backwards. Shouldn't this be `(num ROIs, num images, num images)`?  Same below in the `Returns` section

Comment (ericdill) [inline skbeam/core/correlation.py]: After in-person discussion with @sameera2004 , we have concluded that the two-time correlation results, that is `g2`, should have a shape of `(num ROI, num images, num images)` which will require touching the `_init_state_two_time()` function, and the `two_time_state_to_results()` function and probably the tests and the examples.... (sorry :cry: )

**Comment by sameera2004** [comment]
@ericdill I changed the `two-time correlation results`, that is `g2`, to have a shape of `(num ROI, num images, num images)` and fixed the tests too. They are passing and this PR is ready for review.

File: skbeam/core/tests/test_correlation.py [hunk 2]
```diff
@@ -44,9 +44,12 @@
                                      auto_corr_scat_factor,
                                      lazy_one_time,
                                      lazy_two_time, two_time_corr,
-                                     two_time_state_to_results)
+                                     two_time_state_to_results,
```
Comment (ericdill) [inline skbeam/core/tests/test_correlation.py]: This is imported twice

**Comment by ericdill** [comment]
Did you consider having the one time from two time function return the same thing as `multi_tau_auto_corr`?

**Comment by ericdill** [comment]
Err have it return the correlation and the lag steps?

**Comment by ericdill** [comment]
Oh wait you already have that info from the two time results

**Comment by sameera2004** [comment]
@ericdill No `one_time_from_two_time_function` will not return the same results as `multi_tau_auto_corr`. If you `brute force`(not `multi tau`) using `multi tau auto corr` the results will have same shape.

**Comment by sameera2004** [comment]
@ericdill I fixed the PR ready to review

File: skbeam/core/tests/test_correlation.py [hunk 3]
```diff
@@ -198,6 +200,28 @@ def test_bad_images():
...
+    y_dim = 10
+    stack = 10
+    imgs = np.random.randint(1, 3, (stack, x_dim, y_dim))
+    roi = np.zeros_like(imgs[0])
+    # make sure that the ROIs can be any integers greater than 1.
+    # They do not have to start at 1 and be continuous
+    roi[0:x_dim//10, 0:y_dim//10] = 5
+    roi[x_dim//10:x_dim//5, y_dim//10:y_dim//5] = 3
+
+    two_time = two_time_corr(roi, imgs, stack,
```
Comment (ericdill) [inline skbeam/core/tests/test_correlation.py]: It is a little difficult to understand exactly what `two_time[0]` means in this test, as you need to understand trace the results of `two_time_corr` through about 3 function calls to understand what exactly it returns.  Having just realized that it is difficult to figure out what `two_time[0]` is, I just put in #407 to update the docs for `two_time_corr`.  Could you also change `two_time` to be `g2, lag_steps, _state` to make the meaning of the return value of `two_time_corr` a little clearer in this test?

**Comment by ericdill** [comment]
Hi, one last minor comment. Then it is good to go!

**Comment by sameera2004** [comment]
@ericdill i fixed this PR ready for review. For your PR #407 I left a minor comment, other than it is good to go.

**Comment by ericdill** [comment]
Awesome, thanks @sameera2004 !

**Comment by sameera2004** [comment]
@ericdill Thank you very much

**Pull request merged**